superset/db_engine_specs/elasticsearch.py

```diff
@@ -46,3 +46,18 @@ def convert_dttm(cls, target_type: str, dttm: datetime) -> Optional[str]:
         if target_type.upper() == utils.TemporalType.DATETIME:
             return f"""CAST('{dttm.isoformat(timespec="seconds")}' AS DATETIME)"""
         return None
+
+
+class OpenDistro(ElasticSearchEngineSpec):
+
+    _time_grain_expressions = {
+        None: "{col}",
+    }
+
+    engine = "odelasticsearch"
+    engine_name = "ElasticSearch"
+
+    @classmethod
+    def make_label_compatible(cls, label: str) -> str:
+        new_label = super().make_label_compatible(label)
+        return new_label.replace(".", "_")
```

Comment:
Instead of overriding `make_label_compatible`, we should define a custom `_mutate_label` method that does this. See e.g. https://github.com/apache/superset/blob/9771b82c947193f533ea0720c212175135118daa/superset/db_engine_specs/bigquery.py#L113-L135

Reply:
thks! I guess we should rename `_mutate_label`  to `mutate_label`. This is still on it's infancy, still missing the dbapi PR merge and release and on this side the time grains are different for opendistro

Reply:
Agree on renaming (let's have a separate PR for that). I guess we should have a separate spec for `elasticsearch` and `odelasticsearch` to properly support the differing time grains?